src/Microsoft.Performance.Toolkit.Engine/Engine.cs

```diff
@@ -994,11 +995,24 @@ private RuntimeExecutionResults ProcessCore()
                 {
                     try
                     {
+                        /*
+                         * TODO (#214): for now, the engine ignores the groups returned by processing sources that implement
+                         * IDataSourceGrouper. In the future, this needs to
+                         * 1. Check if the processing source implements IDataSourceGrouper (which it must after V2)
+                         * 2. If not, do what we're doing now and just create a new DataSourceGroup with the default
+                         *     processing mode
+                         * 3. If yes, use an IDataSourceGroupResolver to pick out which of the IDataSourceGroup(s)
+                         *    returned by the processing source should be used for processing.
+                         *      a. If the engine user does not supply a resolver, a sensible one should be used. E.g.
+                         *         one that picks groups that maximize the number of data sources processed without a
+                         *         data source being processed in more than 1 group
```

Comment:
i'm not sure 3.a is true... may be something we have to discuss, but this behavior might lead to misleading data for the user.

it might be better to simply throw an exception and force the host process to provide a grouper in this case.

Reply:
Yeah, that was just an example. We will need to decide this before implementing #214